praw/models/reddit/mixins/__init__.py

```diff
@@ -139,6 +139,61 @@ def remove(self, spam=False):
         data = {"id": self.thing.fullname, "spam": bool(spam)}
         self.thing._reddit.post(API_PATH["remove"], data=data)
 
+    def removal_reasons(self):
+        """Return a dict of available removal reasons.
+
+        Reason ids are required in order to use :meth:`.add_removal_reason`.
+
+        The dict contains two keys: "data" and "order"
+        - The "data" key is a dict of reasons with ID as the key,
+        and "message", "id", and "title" as keys in the sub-dict.
```

Comment:
In my testing, this method returns only the dict of reasons with ID as the key and "message," "id," and "title" as keys in the sub-dict. For example (from my testing subreddit),

```
In [8]: submission.mod.removal_reasons()
Out[8]: 
{'13rr1r9ejdlc7': {'id': '13rr1r9ejdlc7',
  'message': "you're 42 years old",
  'title': '42 removal reason'}}
```

I suspect [the objector (`praw/objector.py`)](https://github.com/praw-dev/praw/blob/73aec2d316eb95306ec537449e6da656fa235211/praw/objector.py) is seeing the `data` key and returning just the item associated with it, because that's a common behavior for much of the rest of the API.

For what it's worth, your tests expect a form like the one I shared above, not the kind described in documentation. So at the moment, tests and implementation agree with each other but disagree with the documentation. Before merging, they should all three agree. I think that ideally the implementation and tests should be changed to match the documentation, not the other way around. If you go that route, make sure to update examples in the documentation to match the changes.

Reply:
So the endpoint will actually return a dict matching what I previously described: a dict with two keys; one for the display order and another with the actual data, but in my implementation and integration test I chose to exclude the "order" list and just return the data portion since I did not feel the order was relevant when using this programmatically.  I updated the documentation to reflect the implementation, but if you feel the order is important to the response I can add it back. Essentially I was trying to make the response easier to work with.